.github/workflows/test-nginx-core.yml

```diff
@@ -0,0 +1,127 @@
+name: test nginx core
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-and-test:
+   runs-on: "ubuntu-20.04"
+   strategy:
+     fail-fast: false
+     matrix:
+       compiler:
+         - { compiler: GNU,  CC: gcc,  CXX: g++}
+         - { compiler: LLVM, CC: clang, CXX: clang++}
+   steps:
+     - uses: actions/checkout@v3
+     - name: get dependencies
+       run: |
+         sudo apt update
+         sudo apt remove nginx libgd3
+         sudo apt install -y libgd-dev libgeoip-dev libxslt1-dev libpcre++0v5 libpcre++-dev liblua5.1-0-dev lua5.1 libperl-dev cpanminus libssl-dev
+         # for building nginx core
+         sudo apt install -y libgoogle-perftools-dev
+         # for running cases in nginx-tests
+         sudo apt install -y uwsgi-plugin-python3 uwsgi ffmpeg memcached libsofthsm2-dev
+     - name: 'checkout luajit2'
+       uses: actions/checkout@v3
+       with:
+         repository: openresty/luajit2
+         path: luajit2
+     - name: 'build luajit2'
+       working-directory: luajit2
+       run: |
+         make
+         sudo make install
+     - name: build
+       env:
+         CC: ${{ matrix.compiler.CC }}
+         CXX: ${{ matrix.compiler.CXX }}
+         LUAJIT_LIB: /usr/local/lib
+         LUAJIT_INC: /usr/local/include/luajit-2.1
+       run: |
+         # TODO: fix https://github.com/alibaba/tengine/issues/1720, then remove "-D T_NGX_HTTP_IMAGE_FILTER=0"
+
+         # NOTE:
+         # For "-D T_NGX_MODIFY_DEFAULT_VALUE=0", we dont compile the source included in this macro, otherwise some nginx-tests cases tests will fail.
+         ./configure  \
+            --with-cc-opt="-D T_NGX_MODIFY_DEFAULT_VALUE=0 -D T_NGX_HTTP_IMAGE_FILTER=0" \
```

Comment:
1. off T_NGX_MODIFY_DEFAULT_VALUE for that tengine modified some default value of directive, such as `worker_processes n;`
2. off T_NGX_HTTP_IMAGE_FILTER for not resolved bug: https://github.com/alibaba/tengine/issues/1720